addons/l10n_tw_edi_ecpay/models/account_move.py

```diff
@@ -0,0 +1,392 @@
+
+    # ----------------
+    # Business methods
+    # ----------------
+    def _get_mail_thread_data_attachments(self):
+        res = super()._get_mail_thread_data_attachments()
+        return res | self.l10n_tw_edi_file_id
+
+    # API methods
+
+    # Check the tax type and special tax type on the invoice lines
+    def _l10n_tw_edi_check_tax_type_on_invoice_lines(self):
+        product_lines = self.invoice_line_ids.filtered(lambda line: line.display_type == "product")
+
+        # Invoive lines without tax or having multiple taxes are not allowed
+        if product_lines.filtered(lambda line: not line.tax_ids or len(line.tax_ids) > 1):
+            raise UserError(self.env._("Invoive lines without tax or having multiple taxes are not allowed."))
+
+        # Create a set of tax types on invoice lines to check if there are multiple tax types or specific tax type on the invoice lines
+        invoice_lines_tax_types = {line.tax_ids[0].l10n_tw_edi_tax_type for line in product_lines if line.tax_ids and line.tax_ids[0].l10n_tw_edi_tax_type}
+
+        if invoice_lines_tax_types:
+            # Tax type "4" is a special tax rate, cannot be mixed with other tax types
+            if len(invoice_lines_tax_types) > 1 and "4" in invoice_lines_tax_types:
+                raise UserError(self.env._("Special tax rate cannot be mixed with other tax types. Please check your tax types on invoice lines."))
+            if "4" in invoice_lines_tax_types and len({line.tax_ids[0].l10n_tw_edi_special_tax_type for line in product_lines if line.tax_ids and line.tax_ids[0].l10n_tw_edi_tax_type == "4" and line.tax_ids[0].l10n_tw_edi_special_tax_type}) > 1:
+                raise UserError(self.env._("Special tax type cannot be mixed with other special tax types. Please check your special tax types on invoice lines."))
+
+            if "2" in invoice_lines_tax_types and "3" in invoice_lines_tax_types:
+                raise UserError(self.env._("Tax type 2 (Zero tax rate) and type 3 (Duty free) cannot be used together. Please check your tax types on invoice lines."))
+        else:
+            raise UserError(self.env._("Please fill in the tax on the invoice lines and select the Ecpay tax type for taxes."))
+
+    # Calculate and return the tax type, special tax type and is zero tax rate included based on the taxes on invoice lines
+    def _l10n_tw_edi_compute_tax_type_from_invoice_lines(self):
+        product_lines = self.invoice_line_ids.filtered(lambda line: line.display_type == "product")
+        # Create a set of tax types on invoice lines to check if there are multiple tax types or specific tax type on the invoice lines
+        invoice_lines_tax_types = {line.tax_ids[0].l10n_tw_edi_tax_type for line in product_lines if line.tax_ids and line.tax_ids[0].l10n_tw_edi_tax_type}
+
+        if invoice_lines_tax_types:
+            if len(invoice_lines_tax_types) > 1:  # mixed tax types
+                return ["9", "0", "2" in invoice_lines_tax_types]
```

Comment:
Please make three variables, set them in the ifs and do a single return at the end.
It will be easier to read and understand.